exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetIsPredicate.java

```diff
@@ -62,70 +62,73 @@ private ParquetIsPredicate(LogicalExpression expr, BiPredicate<Statistics<C>, Ra
-      private final boolean isArray = isArray(expr);
-
-      private boolean isArray(LogicalExpression expression) {
-        if (expression instanceof TypedFieldExpr) {
-          TypedFieldExpr typedFieldExpr = (TypedFieldExpr) expression;
-          SchemaPath schemaPath = typedFieldExpr.getPath();
-          return schemaPath.isArray();
-        }
-        return false;
-      }
-
-      @Override
-      public boolean canDrop(RangeExprEvaluator<C> evaluator) {
+      (exprStat, evaluator) -> {
         // for arrays we are not able to define exact number of nulls
         // [1,2,3] vs [1,2] -> in second case 3 is absent and thus it's null but statistics shows no nulls
-        return !isArray && super.canDrop(evaluator);
-      }
-    };
+        if (expr instanceof TypedFieldExpr) {
+          TypedFieldExpr typedFieldExpr = (TypedFieldExpr) expr;
+          if (typedFieldExpr.getPath().isArray()) {
+            return RowsMatch.SOME;
+          }
+        }
+        if (hasNoNulls(exprStat)) {
+          return RowsMatch.NONE;
+        }
+        return isAllNulls(exprStat, evaluator.getRowCount()) ? RowsMatch.ALL : RowsMatch.SOME;
+      });
   }
 
   /**
    * IS NOT NULL predicate.
    */
   private static <C extends Comparable<C>> LogicalExpression createIsNotNullPredicate(LogicalExpression expr) {
     return new ParquetIsPredicate<C>(expr,
-        //if there are all nulls  -> canDrop
-        (exprStat, evaluator) -> isAllNulls(exprStat, evaluator.getRowCount())
+      (exprStat, evaluator) -> isAllNulls(exprStat, evaluator.getRowCount()) ? RowsMatch.NONE : checkNull(exprStat)
     );
   }
 
   /**
    * IS TRUE predicate.
    */
   private static LogicalExpression createIsTruePredicate(LogicalExpression expr) {
-    return new ParquetIsPredicate<Boolean>(expr, (exprStat, evaluator) ->
-        //if max value is not true or if there are all nulls  -> canDrop
-        isAllNulls(exprStat, evaluator.getRowCount()) || exprStat.hasNonNullValue() && !((BooleanStatistics) exprStat).getMax()
-    );
+    return new ParquetIsPredicate<Boolean>(expr, (exprStat, evaluator) -> {
+      if (isAllNulls(exprStat, evaluator.getRowCount()) || exprStat.hasNonNullValue() && !((BooleanStatistics) exprStat).getMax()) {
+        return RowsMatch.NONE;
+      }
+      return ((BooleanStatistics) exprStat).getMin() ? checkNull(exprStat) : RowsMatch.SOME;
+    });
   }
 
   /**
    * IS FALSE predicate.
    */
   private static LogicalExpression createIsFalsePredicate(LogicalExpression expr) {
     return new ParquetIsPredicate<Boolean>(expr, (exprStat, evaluator) ->
-        //if min value is not false or if there are all nulls  -> canDrop
-        isAllNulls(exprStat, evaluator.getRowCount()) || exprStat.hasNonNullValue() && ((BooleanStatistics) exprStat).getMin()
+      exprStat.hasNonNullValue() && ((BooleanStatistics) exprStat).getMin() && ((BooleanStatistics) exprStat).getMax() || isAllNulls(exprStat, evaluator.getRowCount()) ?
```

Comment:
Please check the condition for "is true": in case all values are `null` NONE is returned without checking for min and/or max. Why is it necessary to AND (`&&`) all nulls condition with "min is true" condition in case of "is false"? Please double check "is true", "is false", "is not true" and "is not false" predicates.

Reply:
It will be good (optional but highly recommended) to add unit test(s) that relies on expression/statistics mocking (not an integration test that relies on parquet files) for ParquetIsPredicate to verify that `canDrop` returns correct result in case of (min:false, max:false, nulls:0), (min:false, max:false, nuls:1), (min:false, max:true, nuls:0), ... (min:true, max:true, nulls:1).

Reply:
Added 4 new tests exec/java-exec/src/test/resources/parquetFilterPush/tfTbl/
ff1 (false, false, 1 null)
ft0 (false, true, 0 null)
tt1 (true, true, 1 null)
Other cases are already tested in exec/java-exec/src/test/resources/parquetFilterPush/blnTbl